src/language-js/needs-parens.js

```diff
@@ -890,6 +890,12 @@ function needsParens(path, options) {
           parent.type !== "VariableDeclarator" &&
           parent.type !== "YieldExpression")
       );
+
+    case "TSInstantiationExpression":
+      return (
+        isMemberExpression(parent) &&
+        node.typeParameters?.type === "TSTypeParameterInstantiation"
```

Comment:
MemberExpression has two children.

**Prettier pr-14701**
[Playground link](https://deploy-preview-14701--prettier.netlify.app/playground/#N4Igxg9gdgLgprEAuEAKAggJ0wQwJ4A8AzjJgJZQDmAfAJQB0OA3ADpQ4DaG2+xpFNWgF02IADQgIABxhloRZKBzYIAdwAKyhApQ4ANqvwKJAI1xgA1nBgBlHAFs4AGQpxkAM31E4p81dtSOGACyKQArj4g3vZkoZgREkQCenAAimEQ8B5ekQBWRAAeNslpGVlInnreEgCOZXDqmNI6IDhEALRQcHAAJr3iIKQ4ZHoCAMIQ9vY4yK16egNJVCnoMPwmYfDqcJguXdlVkQAWMPZ6AOpHZPBEgWBwNtrXZABu13izYETGIC8RAJJQPqwGxgcgydBAmwwPApA7VEBSJrec64KSzJFwbyYF5uCQUbEwRo4SjTeGRQKYbGzGFSLFgsgyAZIigwc5kHowI7IAAcAAYJJg4HUyELiaSZhUchIYDgTOzOdykAAmCRhbwAFTlOkqCLg9hMvT6PScOCoYRJcAAYhBMNM1iFdJsICAAL6uoA)
<!-- prettier-ignore -->
```sh
--parser typescript
```

**Input:**
<!-- prettier-ignore -->
```tsx
(Array<string>).a;
a[(Array<string>)]

```

**Output:**
<!-- prettier-ignore -->
```tsx
(Array<string>).a;
a[(Array<string>)];

```

Reply:
Thanks for suggestion.
I have updated the logic, so it keeps parens only if parent's property is accessed.
So, we keep parens if there is any parent's `property` object except `property` with type `TSInstantiationExpression`.